Tests/Common/test_segmentation_configs.py

```diff
@@ -0,0 +1,364 @@
+
+
+def test_head_and_neck_base() -> None:
+    """
+    Check we can instantiate HeadAndNeckBase class.
+    """
+    ground_truth_ids = DEFAULT_HEAD_AND_NECK_GROUND_TRUTH_IDS
+    config = HeadAndNeckBase(ground_truth_ids)
+    assert config.ground_truth_ids == ground_truth_ids
+
+
+def test_head_and_neck_base_with_optional_params() -> None:
+    """
+    Check that optional parameters can be passed in to HeadAndNeckBase class.
+    """
+    ground_truth_ids = DEFAULT_HEAD_AND_NECK_GROUND_TRUTH_IDS
+    ground_truth_count = len(ground_truth_ids)
+    ground_truth_ids_display_names = generate_random_display_ids(ground_truth_count)
+    colours = generate_random_colours_list(RANDOM_COLOUR_GENERATOR, ground_truth_count)
+    fill_holes = generate_random_fill_holes(ground_truth_count)
+    class_weights = generate_random_class_weights(ground_truth_count + 1)
+    num_feature_channels = random.randint(1, ground_truth_count)
+    slice_exclusion_rules = generate_random_slice_exclusion_rules(ground_truth_ids)
+    summed_probability_rules = generate_random_summed_probability_rules(ground_truth_ids)
+    config = HeadAndNeckBase(
+        ground_truth_ids=ground_truth_ids,
+        ground_truth_ids_display_names=ground_truth_ids_display_names,
+        colours=colours,
+        fill_holes=fill_holes,
+        class_weights=class_weights,
+        slice_exclusion_rules=slice_exclusion_rules,
+        summed_probability_rules=summed_probability_rules,
+        num_feature_channels=num_feature_channels)
+    assert config.ground_truth_ids == ground_truth_ids
+    assert config.ground_truth_ids_display_names == ground_truth_ids_display_names
+    assert config.colours == colours
+    assert config.fill_holes == fill_holes
+    assert config.class_weights == class_weights
+    assert config.feature_channels == [num_feature_channels]
+    assert config.slice_exclusion_rules == slice_exclusion_rules
+    assert config.summed_probability_rules == summed_probability_rules
+
+
+def test_head_and_neck_base_with_invalid_slice_exclusion_rule() -> None:
+    """
+    Check that an invalid slice exclusion rule raises an Exception.
+    """
+    ground_truth_ids = DEFAULT_HEAD_AND_NECK_GROUND_TRUTH_IDS
+    slice_exclusion_rules = [SliceExclusionRule("brainstem2", "spinal_cord", False)]
+    with pytest.raises(Exception):
+        _ = HeadAndNeckBase(
+            ground_truth_ids=ground_truth_ids,
+            slice_exclusion_rules=slice_exclusion_rules)
+
+
+def test_head_and_neck_base_with_invalid_slice_exclusion_rule2() -> None:
+    """
+    Check that an invalid slice exclusion rule raises a Exception.
+    """
+    ground_truth_ids = DEFAULT_HEAD_AND_NECK_GROUND_TRUTH_IDS
+    slice_exclusion_rules = [SliceExclusionRule("brainstem", "spinal_cord2", False)]
+    with pytest.raises(Exception):
+        _ = HeadAndNeckBase(
+            ground_truth_ids=ground_truth_ids,
+            slice_exclusion_rules=slice_exclusion_rules)
```

Comment:
I find it very important to check that it fails for the reason you expect. check that the raised exception has the expected message.
`_ =...` is very F#y, but not sure it's pythonic?